webrender/src/renderer.rs

```diff
@@ -780,7 +835,16 @@ fn create_clip_shader(name: &'static str, device: &mut Device) -> Result<Program
 
     debug!("ClipShader {}", name);
 
-    device.create_program(name, &prefix, &DESC_CLIP)
+    let program = device.create_program(name, &prefix, &DESC_CLIP);
+
+    if let Ok(ref program) = program {
+        device.bind_shader_sampler(program, "sColor0", TextureSampler::Color0);
+        device.bind_shader_sampler(program, "sLayers", TextureSampler::Layers);
+        device.bind_shader_sampler(program, "sRenderTasks", TextureSampler::RenderTasks);
+        device.bind_shader_sampler(program, "sResourceCache", TextureSampler::ResourceCache);
```

Comment:
for cases like this where you need a subset of the table, we can either have 2 separate tables or group the items so that it's easy to select a range with slices

Reply:
Changed it to ```bind_shader_samplers``` which takes an array - good idea.